common/src/service/HttpClientHelper.cs

```diff
@@ -919,5 +894,37 @@ public void Dispose()
             Task<Exception> exception = mapToExceptionFunc(response);
             return await exception.ConfigureAwait(false);
         }
+
+        internal static HttpClient CreateDefaultClient(IWebProxy webProxy, Uri baseUri, int connectionLeaseTimeoutMilliseconds)
+        {
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
+#if NETCOREAPP
+            SocketsHttpHandler httpMessageHandler = new SocketsHttpHandler();
+#else
+            HttpClientHandler httpMessageHandler = new HttpClientHandler();
+#endif
+#pragma warning restore CA2000 // Dispose objects before losing scope
+
+#if !NET451 && !NETCOREAPP
+            httpMessageHandler.SslProtocols = TlsVersions.Instance.Preferred;
+            httpMessageHandler.CheckCertificateRevocationList = TlsVersions.Instance.CertificateRevocationCheck;
+#endif
+
+            if (webProxy != DefaultWebProxySettings.Instance)
+            {
+                httpMessageHandler.UseProxy = webProxy != null;
+                httpMessageHandler.Proxy = webProxy;
+            }
+
+            ServicePointHelpers.SetLimits(httpMessageHandler, baseUri, connectionLeaseTimeoutMilliseconds);
```

Comment:
just thinking of the name is, is baseUri really a "limit" that we are setting?

Reply:
In .NET framework, we set the connection lease timeout on a per Uri basis. Admittedly, this reads a bit funny though.